text/0465-init-pointer-arena.md

```diff
@@ -40,12 +40,22 @@ struct Node<'a> {
 
     a_ref.next = Some(b_ref); // (1)
     b_ref.next = Some(a_ref); // (2)
+    
+    // .. more code which uses the data structure
 }
 ```
 
-Initialising the above example in rust is not possible because of two reasons: first a borrowed reference makes the original reference immutable and second because of the conflicting object's lifetime. On line (1) a borrowed pointer is taken from the `b_ref` pointer, which makes the `b_ref` pointer immutable as long as the `a_ref` reference is alive. As the lifetime of `a_ref` extends to the end of the block, `b_ref` is still borrowed and as it is therefore also immutable the assignment in (2) is prevented.
+Initialising the above example in rust is not possible for two reasons: first a borrowed reference makes the original reference immutable and second because of the conflicting object's lifetime. On line (1), a borrowed pointer is taken from the `b_ref` pointer, which makes the `b_ref` pointer immutable so long as the `a_ref` reference is alive. As the lifetime of `a_ref` extends to the end of the block, `b_ref` is still borrowed and as it is therefore also immutable the assignment in (2) is prevented.
 
-Let's assume for now, the two references `a_ref` and `b_ref` are from a new pointer type that we call `&init T` (similar to `&T` and `&mut T`), which allows to take multiple borrows from an reference and still permits mutation. Even then the above example is rejected by the rust type checker, as the used objects lifetime's conflict. The reference `a_ref` has a larger lifetime as the `b_ref` one. As rust requires the lifetime of field references to be larger than the ones of the object itself, the assignment at (1) is not possible. In fact, the lifetime analysis in rust requires references, that are part of a reference cycle, to either have the same lifetime or rely on weak references (which in turn require runtime checks again).
+The key idea we propose is that it is OK to allow a temporary relaxation of the borrowing rules, effectively allowing multiple mutable borrows in order to set up such structures, so long as: 
+  (a) during this relaxed phase, the borrowed references cannot escape the current thread (as this could lead to race conditions).
+  (b) after this relaxed phase, the borrowed references revert to types which are handled by the current type system (with which the resulting heap structure must still be compatible).
+
+More concretely, we introduce a new pointer type that we call `&init T`, with which a second kind of lifetime is associated: its `init-time`. Rather than specifying when the reference will be deallocated (we will explain later how this is handled), a reference's init-time defines the duration of its init status, during which the relaxed borrow rules can be exploited.
```

Comment:
I called the `init-time` `initialisation phase`, but maybe using `time` instead of `phase` is actually better here :)